llm-service/app/ai/indexing/index.py

```diff
@@ -0,0 +1,131 @@
+#  WARRANTIES WITH RESPECT TO THIS CODE, INCLUDING BUT NOT LIMITED TO
+#  IMPLIED WARRANTIES OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY AND
+#  FITNESS FOR A PARTICULAR PURPOSE; (C) CLOUDERA IS NOT LIABLE TO YOU,
+#  AND WILL NOT DEFEND, INDEMNIFY, NOR HOLD YOU HARMLESS FOR ANY CLAIMS
+#  ARISING FROM OR RELATED TO THE CODE; AND (D)WITH RESPECT TO YOUR EXERCISE
+#  OF ANY RIGHTS GRANTED TO YOU FOR THE CODE, CLOUDERA IS NOT LIABLE FOR ANY
+#  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+#  CONSEQUENTIAL DAMAGES INCLUDING, BUT NOT LIMITED TO, DAMAGES
+#  RELATED TO LOST REVENUE, LOST PROFITS, LOSS OF INCOME, LOSS OF
+#  BUSINESS ADVANTAGE OR UNAVAILABILITY, OR LOSS OR CORRUPTION OF
+#  DATA.
+#
+
+from dataclasses import dataclass
+import logging
+import os
+from typing import Dict, List, Type
+
+from .readers.pdf import PDFReader
+from .readers.nop import NopReader
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.node_parser import SentenceSplitter
+from llama_index.core.schema import Document
+from llama_index.core.base.embeddings.base import BaseEmbedding
+from ...services.vector_store import VectorStore
+from llama_index.core.node_parser.interface import BaseNode
+
+logger = logging.getLogger(__name__)
+
+READERS: Dict[str, Type[BaseReader]] = {
+    ".pdf": PDFReader,
+    ".txt": NopReader,
+    ".md": NopReader,
+}
+CHUNKABLE_FILE_EXTENSIONS = set(
+    [
+        ".pdf",
+        ".txt",
+        ".md",
+    ]
+)
+
+@dataclass
+class NotSupportedFileExtensionError(Exception):
+    file_extension: str
+
+class Indexer:
+    def __init__(self, data_source_id: int, splitter: SentenceSplitter, embedding_model: BaseEmbedding, chunks_vector_store: VectorStore):
+        self.data_source_id = data_source_id
+        self.splitter = splitter
+        self.embedding_model = embedding_model
+        self.chunks_vector_store = chunks_vector_store
+
+    def index_file(self, file_path: str, file_id: str):
+        logger.debug(f"Indexing file: {file_path}")
+
+        file_extension = os.path.splitext(file_path)[1]
+        reader_cls = READERS.get(file_extension)
+        if not reader_cls:
+            raise NotSupportedFileExtensionError(file_extension)
+
+        reader = reader_cls()
+
+        logger.debug(f"Parsing file: {file_path}")
+
+        documents = self._documents_in_file(reader, file_path, file_id)
+        if file_extension in CHUNKABLE_FILE_EXTENSIONS:
+            logger.debug(f"Chunking file: {file_path}")
+            chunks = [chunk for document in documents for chunk in self._chunks_in_document(document)]
+        else:
+            chunks = documents
+
+        texts = [chunk.text for chunk in chunks]
+        logger.debug(f"Embedding {len(texts)} chunks")
+        embeddings = self.embedding_model.get_text_embedding_batch(texts)
+
+        for chunk, embedding in zip(chunks, embeddings):
+            chunk.embedding = embedding
```

Comment:
We need to write the doc id and the file name into the chunk:

```suggestion
            chunk.embedding = embedding
            chunk.metadata["document_id"] = file_id
            chunk.metadata["file_name"] = os.path.basename(file_path)
```

Reply:
or probably below when the rest of the chunk metadata is being written...